### PR TITLE
Add API for reregistering a deregistered collector

### PR DESCRIPTION
**Pull request by annyhe**
- add the api endpoint for re-registering a collector, after it has been de-registered.
- includes tests

### CODE REVIEW

File: tests/api/v1/collectors/re-register.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/api/v1/collectors/re-register.js
```
Comment (iamigo) [inline tests/api/v1/collectors/re-register.js]: no dash in file name

File: api/v1/controllers/collectors.js [hunk 2]
```diff
@@ -206,6 +206,37 @@ function deregisterCollector(req, res, next) {
...
 /**
+ * Reregister a collector. Access restricted to Refocus Collector only.
+ *
+ * @param {IncomingMessage} req - The request object
+ * @param {ServerResponse} res - The response object
+ * @param {Function} next - The next middleware function in the stack
+ */
+function reRegisterCollector(req, res, next) {
+
+  //TODO: remove token verification once it's set up in middleware for collectors
```
Comment (shriramshankar) [inline api/v1/controllers/collectors.js]: The middleware has already been setup. Check for req.headers.IsCollector to make sure that it is a collector.

File: api/v1/controllers/collectors.js [hunk 3]
```diff
@@ -206,6 +206,37 @@ function deregisterCollector(req, res, next) {
 } // deregisterCollector
 
 /**
+ * Reregister a collector. Access restricted to Refocus Collector only.
```
Comment (shriramshankar) [inline api/v1/controllers/collectors.js]: Wondering if a user can reregister a collector.?

Comment (annyhe) [reply api/v1/controllers/collectors.js]: What would be the use case? What are your thoughts @iamigo

**Comment by annyhe** [comment]
Updated. Please review @iamigo @shriramshankar

File: api/v1/controllers/collectors.js (hunk 3 above)
Comment (iamigo) [reply api/v1/controllers/collectors.js]: I think just a collector.

**Comment by annyhe** [comment]
updated. Please review @shriramshankar

**Pull request merged**